### PR TITLE
Fix issue 13619 - array capacity not updated when changing length

### PR DESCRIPTION
**Pull request by byebye**

### CODE REVIEW

File: std/container/array.d [hunk 1]
```diff
@@ -333,6 +333,7 @@ if (!is(Unqual!T == bool))
             _payload = (cast(T*) realloc(_payload.ptr,
                             nbytes))[0 .. newLength];
             initializeAll(_payload.ptr[startEmplace .. length]);
```
Comment (JackStouffer) [inline std/container/array.d]: Also, shouldn't this line be

```
initializeAll(_payload.ptr[startEmplace .. newLength]);
```

I think it's a no-op currently.

Comment (byebye) [inline std/container/array.d]: It seems so!

**Comment by JackStouffer** [comment]
If `capacity > length && capacity > newLength` then the `realloc` isn't needed.

This can be done in different PR (as it's an optimization not a bug fix) if you don't want to add it to this one.

**Comment by JackStouffer** [comment]
@byebye Yeah, thinking about it more, that optimization should go into another PR

@wilzbach The CI is failing with a dub error

**Comment by byebye** [comment]
Good idea, I'm gonna add it in another PR as you suggested.

**Pull request merged**